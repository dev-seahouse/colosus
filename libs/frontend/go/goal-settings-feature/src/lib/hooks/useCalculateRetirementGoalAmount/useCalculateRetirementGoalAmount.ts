import { useQuery } from '@tanstack/react-query';
import { BambuApiLibraryIntegrationRetirementApi } from '@bambu/api-client';
import type { CalculateRetirementGoalAmountRequestDto } from '@bambu/api-client';

export const calculateRetirementGoalAmountQuery = (
  args: CalculateRetirementGoalAmountRequestDto,
  enabled: boolean
) => ({
  queryKey: ['calculateRetirementGoalAmount', args],
  queryFn: async () => {
    const bambuApiLibraryIntegrationRetirementApi =
      new BambuApiLibraryIntegrationRetirementApi();
    const res =
      await bambuApiLibraryIntegrationRetirementApi.calculateRetirementGoalAmount(
        args
      );
    return res.data;
  },
  enabled,
});

export interface UseCalculateRetirementGoalAmountOptions {
  enabled?: boolean;
  args: CalculateRetirementGoalAmountRequestDto;
}

/**
 * query hook to calculate retirement goal amount
 */
export function useCalculateRetirementGoalAmount({
  args,
  enabled = true,
}: UseCalculateRetirementGoalAmountOptions) {
  return useQuery({
    ...calculateRetirementGoalAmountQuery(args, enabled),
  });
}
export default useCalculateRetirementGoalAmount;
