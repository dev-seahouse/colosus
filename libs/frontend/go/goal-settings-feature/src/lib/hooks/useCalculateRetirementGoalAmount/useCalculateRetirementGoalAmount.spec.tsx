import {
  renderHook,
  queryClientWrapper,
  waitFor,
} from '@bambu/react-test-utils';
import {
  CalculateRetirementGoalAmountRequestGender,
  CalculateRetirementGoalAmountRequestPeriod,
} from '@bambu/api-client';
import useCalculateRetirementGoalAmount from './useCalculateRetirementGoalAmount';

const args = {
  age: 25,
  annualisedInflationRate: 1.25,
  annualisedSavingsAcctIntR: 0.5,
  annualRetirementIncome: 250000,
  country: 'US',
  gender: CalculateRetirementGoalAmountRequestGender.MALE,
  compoundsPerYear: 12,
  lifeExpectancyFemale: 81,
  lifeExpectancyMale: 80,
  retirementAge: 55,
  period: CalculateRetirementGoalAmountRequestPeriod.END,
};
const enabled = true;

describe('useCalculateRetirementGoalAmount', () => {
  it('should call onSuccess when operation is successful', async () => {
    const { result } = renderHook(
      () =>
        useCalculateRetirementGoalAmount({
          enabled,
          args,
        }),
      {
        wrapper: queryClientWrapper,
      }
    );

    await waitFor(() => expect(result.current.isSuccess).toBe(true));

    expect(result.current.data).toMatchSnapshot();
  });
});
