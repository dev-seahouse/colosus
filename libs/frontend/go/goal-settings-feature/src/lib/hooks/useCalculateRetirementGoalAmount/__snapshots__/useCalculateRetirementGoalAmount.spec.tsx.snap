// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useCalculateRetirementGoalAmount > should call onSuccess when operation is successful 1`] = `
{
  "goalAmount": 1612818,
  "retirementAssets": {
    "totalAssets": 443392.3,
    "totalPension": 235294.11,
    "totalRetirementSavings": 12682.42,
    "totalSocialSecurityBenefits": 195415.77,
  },
  "totalRetirementExpenditure": 2056210.3,
  "workingPeriod": 12,
  "yearsToDecumulate": 20,
}
`;
